server/test/deck.test.js

```diff
@@ -233,143 +220,157 @@ describe("DELETE Invalid Decks", function () {
+                          practiceDeckId +
+                          "/cards/" +
+                          res.body["_id"]
+                      )
+                      .end(function (err, res) {
+                        res.should.have.status(200);
+                        done();
+                      });
+                  });
+              });
+          });
+      });
+  });
 
-    it("Restore DB", function (done) {
-      chai
-        .request(server)
-        .delete("/api/decks/" + practiceDeckId)
-        .end(function (err, res) {
-          res.should.have.status(200);
-          done();
-        });
-    });
+  it("good, repetition: 1, interval: 1 => interval: 6", function (done) {
+    let cardId;
+    chai
+      .request(server)
+      .post("/api/decks/" + practiceDeckId + "/cards")
+      .send({
+        front: "front2",
+        back: "back2",
+      })
+      .end(function (err, res) {
+        res.should.have.status(201);
+        cardId = res.body["_id"];
+        chai
+          .request(server)
+          .patch(
+            "/api/decks/" +
+              practiceDeckId +
+              "/cards/" +
+              res.body["_id"] +
+              "/recallability"
+          )
+          .send({ recallability: "good" })
+          .end(function (err, res) {
+            res.should.have.status(200);
+            chai
+              .request(server)
+              .get("/api/decks/" + practiceDeckId + "/practice")
+              .end(function (err, res) {
+                res.should.have.status(200);
+                res.body.should.have.property("repetition").eql(0);
+                res.body.should.have.property("interval").eql(0);
+                chai
+                  .request(server)
+                  .get("/api/decks/" + practiceDeckId + "/practice/next")
+                  .end(function (err, res) {
+                    res.should.have.status(200);
+                    chai
+                      .request(server)
+                      .get("/api/decks/" + practiceDeckId + "/practice/next")
+                      .end(function (err, res) {
+                        res.should.have.status(200);
+                        chai
+                          .request(server)
+                          .get(
+                            "/api/decks/" +
+                              practiceDeckId +
+                              "/cards/" +
+                              cardId
+                          )
+                          .end(function (err, res) {
+                            res.should.have.status(200);
+                            res.body.should.have.property("repetition").eql(2);
+                            res.body.should.have.property("interval").eql(6);
+                            done();
+                          });
+                      });
+                  });
+              });
+          });
+      });
   });
-  this.afterEach(() => {
-    middlewareStub.restore();
```

Comment:
fyi `beforeEach()` at line 245 and above lines deleted were added to make the workflow test pass. I added them since the tests will pass in our local but without these before and after, it gave us an error in the workflow. Just a note in case the tests don't pass again in workflow after you merge.